sklearn/tree/tree.py

```diff
@@ -301,11 +301,13 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                 sample_weight = expanded_class_weight
 
         # Set min_weight_leaf from min_weight_fraction_leaf
-        if self.min_weight_fraction_leaf != 0. and sample_weight is not None:
+        if sample_weight is None:
+            min_weight_leaf = int(ceil(self.min_weight_fraction_leaf *
+                                       n_samples))
+            min_samples_leaf = max(min_samples_leaf, min_weight_leaf)
```

Comment:
I'm not sure whether it's better to have this block as what it is, or
```
min_samples_leaf = max(min_samples_leaf, int(ceil(self.min_weight_fraction_leaf * n_samples)))
min_weight_leaf = 0
```

or simply omit the `min_samples_leaf = max(...)` and just set `min_weight_leaf`.